Sources/SKTestSupport/SwiftPMTestProject.swift

```diff
@@ -33,6 +35,117 @@ public class SwiftPMTestProject: MultiFileTestProject {
     )
     """
 
+  /// A manifest that defines two targets:
+  ///  - A macro target named `MyMacro`
+  ///  - And executable target named `MyMacroClient`
+  ///
+  /// It builds the macro using the swift-syntax that was already built as part of the SourceKit-LSP build.
+  /// Re-using the SwiftSyntax modules that are already built is significantly faster than building swift-syntax in
+  /// each test case run and does not require internet access.
+  public static var macroPackageManifest: String {
+    get async throws {
+      // build directory is `.build/debug/arm64-apple-macosx` or the platform-specific equivalent
+      let buildDirectory = testBundle.bundleURL.deletingLastPathComponent()
+
+      // Directories that we should search for the swift-syntax package.
+      // We prefer a checkout in the build folder. If that doesn't exist, we are probably using local dependencies
+      // (SWIFTCI_USE_LOCAL_DEPS), so search next to the sourcekit-lsp source repo
+      let swiftSyntaxSearchPaths = [
+        buildDirectory
+          .deletingLastPathComponent()  // arm64-apple-macosx
+          .deletingLastPathComponent()  // debug
+          .appendingPathComponent("checkouts"),
+        URL(fileURLWithPath: #filePath)
+          .deletingLastPathComponent()  // SwiftPMTestProject.swift
+          .deletingLastPathComponent()  // SKTestSupport
+          .deletingLastPathComponent()  // Sources
+          .deletingLastPathComponent(),  // sourcekit-lsp
+      ]
+
+      let swiftSyntaxCShimsModulemap =
+        swiftSyntaxSearchPaths.map { swiftSyntaxSearchPath in
+          swiftSyntaxSearchPath
+            .appendingPathComponent("swift-syntax")
+            .appendingPathComponent("Sources")
+            .appendingPathComponent("_SwiftSyntaxCShims")
+            .appendingPathComponent("include")
+            .appendingPathComponent("module.modulemap")
+        }
+        .first { FileManager.default.fileExists(atPath: $0.path) }
+
+      guard let swiftSyntaxCShimsModulemap else {
+        throw SwiftSyntaxCShimsModulemapNotFoundError()
+      }
```

Comment:
Hrm, unfortunate this is needed, shame we don't have a non-deprecated way of using `@_implementationOnly` in a non-resilient module